libgadget/treewalk.h

```diff
@@ -51,6 +51,7 @@ typedef struct {
     int * ngblist;
     int64_t Ninteractions;
     int64_t Nnodesinlist;
+    int64_t Nlist;
```

Comment:
Since we are at this, could you also add a comment about the meaning of each counter?